### PR TITLE
feat(ddm): Implement a new op-based system for changing metrics visibility

### PR DESCRIPTION
**Pull request by iambriccardo**
This PRs implements a way better mechanism to handle blocking of metrics as opposed to before. The previous implementation was faulty because it didn't distinguish between a metric being blocked and only the tags of the metric being blocked. The new system does this differentiation and is built with an op-based approach that makes state changes very easy from the outside.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_metrics.py [hunk 1]
```diff
@@ -67,27 +115,25 @@ def get(self, request: Request, organization) -> Response:
 
         return Response(metrics, status=200)
 
-    def post(self, request: Request, organization) -> Response:
+    def put(self, request: Request, organization) -> Response:
```
Comment (iambriccardo) [inline src/sentry/api/endpoints/organization_metrics.py:118]: Changed to put as suggested by @obostjancic.

File: src/sentry/sentry_metrics/visibility/metrics_blocking.py [hunk 2]
```diff
@@ -14,36 +14,63 @@ class BlockedMetricsRelayConfig(TypedDict):
...
-    def merge(self, other: "BlockedMetric") -> "BlockedMetric":
-        return BlockedMetric(metric_mri=self.metric_mri, tags=self.tags.union(other.tags))
+    def apply(self, other: MetricOperation) -> "BlockedMetric":
+        return BlockedMetric(
+            metric_mri=self.metric_mri,
+            is_blocked=other.block_metric if other.block_metric is not None else self.is_blocked,
+            blocked_tags=self.blocked_tags.union(other.block_tags) - other.unblock_tags,
+        )
+
+    def is_useless(self):
```
Comment (obostjancic) [inline src/sentry/sentry_metrics/visibility/metrics_blocking.py:57]: maybe something like: 
```suggestion
    def is_empty_rule(self):
```

**Review by obostjancic — APPROVED** [review]
Nice!

File: src/sentry/sentry_metrics/visibility/metrics_blocking.py (hunk 2 above)
Comment (iambriccardo) [reply src/sentry/sentry_metrics/visibility/metrics_blocking.py:57]: Will fix in the follow up to not slow down the deployment.

**Pull request merged**